web-ui/src/components/reports-section/CheckinReport.jsx

```diff
@@ -1,36 +1,188 @@
-import React, { useContext } from 'react';
+import React, { useContext, useState } from 'react';
 import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
 
 import { getAvatarURL } from '../../api/api.js';
 import { AppContext } from '../../context/AppContext';
 import { selectFilteredCheckinsForTeamMemberAndPDL } from '../../context/selectors';
 
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import ExpandMore from '../expand-more/ExpandMore';
+
 import {
   Accordion,
   AccordionDetails,
   AccordionSummary,
   Avatar,
   Box,
   Card,
-  CardHeader,
   CardContent,
+  CardHeader,
   Chip,
+  Collapse,
   Container,
+  Divider,
+  Stepper,
+  Step,
+  StepLabel,
   Typography
 } from '@mui/material';
 
+/**
+ * @typedef {Object} Checkin
+ * @property {string} id - The ID of the check-in.
+ * @property {boolean} completed - Indicates whether the check-in is completed.
+ * @property {Array} checkinDate - The date of the check-in.
+ * @property {string} pdlId - The ID of the PDL.
+ * @property {string} teamMemberId - The ID of the team member.
+ */
+
+/**
+ * @typedef {("Done" | "In Progress" | "Not Started")} CheckinStatus
+ * @typedef {("Not Yet Scheduled" | "Scheduled" | "Completed")} SchedulingStatus
+ */
+
+/** @type {CheckinStatus} */
+const steps = ['Not Started', 'In Progress', 'Done'];
+
+/** @type {SchedulingStatus} */
+const schedulingSteps = ['Not Yet Scheduled', 'Scheduled', 'Completed'];
+
+function HorizontalLinearStepper({ step = 0 }) {
+  const [activeStep, setActiveStep] = React.useState(step);
+  const [skipped, setSkipped] = React.useState(new Set());
+
+  const isStepOptional = step => step === -1;
+  const isStepSkipped = step => skipped.has(step);
+
+  return (
+    <Box sx={{ width: '100%', my: 1 }}>
+      <Stepper activeStep={activeStep}>
+        {steps.map((label, index) => {
+          const stepProps = {};
+          const labelProps = {};
+          if (isStepOptional(index)) {
+            labelProps.optional = (
+              <Typography variant="caption">Optional</Typography>
+            );
+          }
+          if (isStepSkipped(index)) {
+            stepProps.completed = false;
+          }
+          return (
+            <Step key={label} {...stepProps}>
+              <StepLabel {...labelProps}>{label}</StepLabel>
+            </Step>
+          );
+        })}
+      </Stepper>
+    </Box>
+  );
+}
+
```

Comment:
Seems odd for the following import to not appear near the top of the file with the other imports.

Reply:
Indeed! Nice catch. Resolved.